### PR TITLE
Subscribe block: button-only design

### PR DESCRIPTION
**Pull request by simison**
Part of https://github.com/Automattic/jetpack/issues/35708

## Proposed changes:

* Button-only design. Email is asked in the followed popup that opens.

Screenshot 2023-11-01 at 19 21 01

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/view.scss [hunk 1]
```diff
@@ -24,6 +24,15 @@
 		border-inline-end-width: 0 !important;
 	}
 }
+.is-style-button {
+	.wp-block-jetpack-subscriptions__textfield {
+		display: none;
+	}
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/view.scss]: Should just not render it in first place, instead of CSS hiding.

Comment (pkuliga) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/view.scss]: Fixed.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js [hunk 2]
```diff
@@ -92,6 +93,8 @@ export function SubscriptionEdit( props ) {
 		successMessage,
 	} = validatedAttributes;
 
+	const isButtonOnlyStyle = validatedAttributes.className === BUTTON_ONLY_CLASS_NAME;
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js]: There's a helper function, example:

https://github.com/Automattic/jetpack/blob/67bcc94e6a60499f3ae900aff7cc6cd4123bb3e2/projects/plugins/jetpack/extensions/blocks/tiled-gallery/edit.js#L268

The className can be multiple classes, including custom ones inputed in editor by user.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 3]
```diff
@@ -423,12 +423,16 @@ function get_element_styles_from_attributes( $attributes ) {
 	$submit_button_styles .= $style;
 	$email_field_styles   .= $style;
 
-	$button_spacing = get_attribute( $attributes, 'spacing', DEFAULT_SPACING_VALUE );
-	if ( true === get_attribute( $attributes, 'buttonOnNewLine' ) ) {
-		$submit_button_styles .= sprintf( 'margin-top: %dpx;', $button_spacing );
-	} else {
-		$submit_button_styles .= 'margin: 0px; '; // Reset Safari's 2px default margin for buttons affecting input and button union
-		$submit_button_styles .= sprintf( 'margin-left: %dpx;', $button_spacing );
+	$is_button_only_style = get_attribute( $attributes, 'className' ) === BUTTON_ONLY_CLASS_NAME;
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:426]: There should be PHP equivalent of the helper I mentioned above https://github.com/Automattic/jetpack/pull/33921/files/16f39dd56ac23f45bcbf557da1839d13faf1d8d5..35beec6f24a4831e79557ed3e04002b4fc8ac0d6#r1489104415

File: projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js (hunk 2 above)
Comment (pkuliga) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js]: Fixed!

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php (hunk 3 above)
Comment (pkuliga) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:426]: I couldn't find it, asked the Crew folks p1707903492410909-slack-CBG1CP4EN.

**Comment by simison** [comment]
Superseded by https://github.com/Automattic/jetpack/pull/37341

**Pull request closed without merging**